src/include/buffer.h

```diff
@@ -818,7 +819,7 @@ namespace buffer CEPH_BUFFER_API {
 
     // clone non-shareable buffers (make shareable)
     void make_shareable() {
-      std::list<buffer::ptr>::iterator pb;
+      decltype(_buffers)::iterator pb;
```

Comment:
i assume we have access to `buffers_t` at this point? if yes, why not use `buffers_t` ?

Reply:
I don't see any reason for `decltype` now. I believe it's just an artifact from development process. Switching to `buffers_t`.

Reply:
@tchaikov: just pushed a commit switching `buffers.h` to range-based for loop instead of the `decltype` and iterator construct.